lib/new_relic/agent/new_relic_service.rb

```diff
@@ -436,21 +440,19 @@ def relay_request(request, opts)
       end
 
       def attempt_request(request, opts)
-        response = nil
         conn = http_connection
         ::NewRelic::Agent.logger.debug("Sending request to #{opts[:collector]}#{opts[:uri]} with #{request.method}")
-        Timeout.timeout(@request_timeout) do
-          response = conn.request(request)
-        end
-        response
+        conn.request(request)
       end
 
       def handle_error_response(response, endpoint)
         case response
         when Net::HTTPRequestTimeOut,
              Net::HTTPTooManyRequests,
              Net::HTTPInternalServerError,
-             Net::HTTPServiceUnavailable
+             Net::HTTPServiceUnavailable,
+             Net::OpenTimeout,
+             Net::ReadTimeout
```

Comment:
In the method `setup_connection_timeouts` that you updated, I saw that one of the new things there was an ssl_timeout being set as well. Is that something we would need to handle here as well?

Reply:
Great question!!! Net::HTTP itself will only raise 1 of 3 error types for open, read, and write. But when setting the timeout values, it supports an "ssl" type, so shouldn't there be a corresponding error for that ssl timeout being hit? I thought the same thing and forgot to proactively bring it up here in the PR.

My current understanding: Net::HTTP simply passes the ssl timeout to the underlying OpenSSL::SSL::SSLContext class. If that class has a timeout error, then something ssl specific has gone wrong and I'm currently thinking we want to let the `else` flow of the `case` statement handle it. Let me know if you have other thoughts. One idea I had was to get out of the business of setting the ssl timeout anywhere in the first place and just using OpenSSL's presumably reasonable defaults.

Reply:
Removed the Net::HTTP specific SSL timeout with [607f6d5](https://github.com/newrelic/newrelic-ruby-agent/pull/2147/commits/607f6d5c9466bffb725d9151205fd411637f9ad9)